clang/lib/AST/Decl.cpp

```diff
@@ -3729,6 +3729,11 @@ unsigned FunctionDecl::getBuiltinID(bool ConsiderWrapperFunctions) const {
       !(BuiltinID == Builtin::BIprintf || BuiltinID == Builtin::BImalloc))
     return 0;
 
+  if (Context.getLangOpts().isSYCL() && hasAttr<SYCLDeviceOnlyAttr>() &&
+      !(BuiltinID == Builtin::BIprintf || BuiltinID == Builtin::BImalloc)) {
```

Comment:
If I get it right, we allow `printf` and `malloc` for **all** SYCL targets. I'm open to discuss if requiring `printf` support is reasonable, but I'm not sure if `malloc` can be supported by all SYCL targets.

Reply:
I think this is a copy/paste from the CUDA code just above, malloc shouldn't be there